bwapi/Library/BWAPILIBUnitTest/UnitTypeTest.cpp

```diff
@@ -0,0 +1,1895 @@
+    EXPECT_EQ(UnitTypes::Powerup_Young_Chrysalis.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Psi_Emitter, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Psi_Emitter.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Data_Disk, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Data_Disk.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Khaydarin_Crystal, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Khaydarin_Crystal.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Mineral_Cluster_Type_1, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Mineral_Cluster_Type_1.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Mineral_Cluster_Type_2, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Mineral_Cluster_Type_2.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Protoss_Gas_Orb_Type_1, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Protoss_Gas_Orb_Type_1.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Protoss_Gas_Orb_Type_2, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Protoss_Gas_Orb_Type_2.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Zerg_Gas_Sac_Type_1, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Zerg_Gas_Sac_Type_1.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Zerg_Gas_Sac_Type_2, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Zerg_Gas_Sac_Type_2.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Terran_Gas_Tank_Type_1, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Terran_Gas_Tank_Type_1.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Powerup_Terran_Gas_Tank_Type_2, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Powerup_Terran_Gas_Tank_Type_2.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(None, whatBuilds) {
+    EXPECT_EQ(UnitTypes::None.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(AllUnits, whatBuilds) {
+    EXPECT_EQ(UnitTypes::AllUnits.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Men, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Men.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Buildings, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Buildings.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Factories, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Factories.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Unknown, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Unknown.whatBuilds(), std::make_pair(UnitTypes::Unknown, 1));
+}
+
+TEST(Terran_Marine, requiredUnits) {
+    std::map<UnitType, int> rUnitMap{ {UnitTypes::Terran_Barracks, 1} };
+    EXPECT_EQ(UnitTypes::Terran_Marine.requiredUnits(), rUnitMap);
+}
+
+TEST(Terran_Ghost, requiredUnits) {
+    std::map<UnitType, int> rUnitMap{ {UnitTypes::Terran_Barracks, 1},  {UnitTypes::Terran_Academy, 1}, {UnitTypes::Terran_Covert_Ops, 1} };
+    EXPECT_EQ(UnitTypes::Terran_Ghost.requiredUnits(), rUnitMap);
+}
```

Comment:
Try [UnorderedElementsAre](https://google.github.io/googletest/gmock_cheat_sheet.html#container-matchers). This should give a better error message for a test failure (i.e. which items are missing, which items are extra).

```cpp
using BWAPI::UnitTypes;

std::pair<UnitType, int> One(UnitType t) {
  return std::make_pair(t, 1);
}

TEST(Terran_Ghost, requiredUnits) {
  EXPECT_THAT(Terran_Ghost.requiredUnits(), UnorderedElementsAre(One(Terran_Barracks), One(Terran_Academy), One(Terran_Covert_Ops)));
}
```